src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -39,5 +40,22 @@ public override Analyzer CreateAnalyzer(ISyntaxFacts syntaxFacts, ParseOptions o
                 (version >= LanguageVersion.CSharp9 ? Feature.RelationalPattern | Feature.OrPattern | Feature.AndPattern | Feature.TypePattern : 0);
             return new CSharpAnalyzer(syntaxFacts, features);
         }
+
+        protected override SyntaxTriviaList GetLeadingTriviaToTransfer(SyntaxNode syntaxToRemove)
+        {
+            if (syntaxToRemove is IfStatementSyntax { Parent: ElseClauseSyntax elseClause } &&
```

Comment:
does this work if we have a comment above an else? If so, is there a test for it?

```csharp
if (x == 0)
{
}
// comment
else if (x == 1)
{
}
// comment
else
{
}
```

Reply:
i'll add that :)